File: day-01-foundations/module-02-producer/lab-1.2c-producer-error-handling/EBankingResilientProducerAPI/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace EBankingResilientProducerAPI.Models;

public enum TransactionType
{
    Transfer = 1,
    Payment = 2,
    Deposit = 3,
    Withdrawal = 4,
    Refund = 5,
    International = 6
}

public enum TransactionStatus
{
    Pending,
    Processing,
    Completed,
    Failed,
    Rejected
}

public class Transaction
{
    [Required]
    public string TransactionId { get; set; } = Guid.NewGuid().ToString();

    [Required]
    [StringLength(20, MinimumLength = 10)]
    public string FromAccount { get; set; } = string.Empty;

    [Required]
    [StringLength(20, MinimumLength = 10)]
    public string ToAccount { get; set; } = string.Empty;

    [Required]
    [Range(0.01, 1_000_000.00)]
    public decimal Amount { get; set; }

    [Required]
    [StringLength(3, MinimumLength = 3)]
    public string Currency { get; set; } = "EUR";

    [Required]
    public TransactionType Type { get; set; }

    [StringLength(500)]
    public string? Description { get; set; }

    [Required]
    public string CustomerId { get; set; } = string.Empty;

    public DateTime Timestamp { get; set; } = DateTime.UtcNow;

    [Range(0, 100)]
    public int RiskScore { get; set; } = 0;

    public TransactionStatus Status { get; set; } = TransactionStatus.Pending;
}

File: day-02-development/module-04-advanced-patterns/lab-2.3a-consumer-dlt-retry/java/src/main/java/com/data2ai/kafka/dltretry/model/Transaction.java
package com.data2ai.kafka.dltretry.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Transaction {
    private String transactionId;
    private String customerId;
    private String fromAccount;
    private String toAccount;
    private BigDecimal amount;
    private String currency;
    private TransactionType type;
    private TransactionStatus status;
    private Instant timestamp;
    private String description;
    private Map<String, Object> metadata;

    public enum TransactionType {
        WITHDRAWAL, PAYMENT, CARD_PAYMENT, TRANSFER,
        INTERNATIONAL_TRANSFER, BILL_PAYMENT, DEPOSIT
    }

    public enum TransactionStatus {
        PENDING, COMPLETED, FAILED, CANCELLED
    }
}
